### PR TITLE
qa: Prevent concurrency issues reading .cookie file

### LINKED ISSUE #15733: wallet_multiwallet test failed with BerkeleyBatch: Can't open database, duplicates fileid
Error thrown once:

```
BerkeleyBatch: Can't open database w8_copy (duplicates fileid 595e23002f000000b5cefe1eff76000000000000 from w8)
```

In https://travis-ci.org/bitcoin/bitcoin/jobs/515252355#L4924

### PR DESCRIPTION
**Pull request by promag**
Hopefully fixes #15733.

### CODE REVIEW

File: test/functional/test_framework/util.py [hunk 1]
```diff
@@ -323,12 +330,13 @@ def get_auth_cookie(datadir):
...
                     assert password is None  # Ensure that there is only one rpcpassword line
                     password = line.split("=")[1].strip("\n")
-    if os.path.isfile(os.path.join(datadir, "regtest", ".cookie")) and os.access(os.path.join(datadir, "regtest", ".cookie"), os.R_OK):
-        with open(os.path.join(datadir, "regtest", ".cookie"), 'r', encoding="ascii") as f:
-            userpass = f.read()
-            split_userpass = userpass.split(':')
-            user = split_userpass[0]
-            password = split_userpass[1]
+    cookie = os.path.join(datadir, "regtest", ".cookie")
+    wait_until(lambda: can_read_file(cookie), timeout=5)
```
Comment (MarcoFalke) [inline test/functional/test_framework/util.py]: I believe this method is not allowed to block, it must return a result immediately

Can be fixed by keeping the previous `if`-logic

Comment (promag) [reply test/functional/test_framework/util.py]: > I believe this method is not allowed to block, it must return a result immediately

Why not?

Comment (MarcoFalke) [reply test/functional/test_framework/util.py]: See `wait_for_rpc_connection`, which calls this method

Comment (promag) [reply test/functional/test_framework/util.py]: Thanks, `wait_for_rpc_connection` already retries so I've changed this remove the early checks and just try opening the cookie file.

**Comment by MarcoFalke** [comment]
utACK cc3bee31cc8eeeba7ab2e0f3d53b004d03fb1ab9

File: test/functional/test_framework/util.py [hunk 2]
```diff
@@ -323,12 +323,14 @@ def get_auth_cookie(datadir):
...
                     assert password is None  # Ensure that there is only one rpcpassword line
                     password = line.split("=")[1].strip("\n")
-    if os.path.isfile(os.path.join(datadir, "regtest", ".cookie")) and os.access(os.path.join(datadir, "regtest", ".cookie"), os.R_OK):
+    try:
         with open(os.path.join(datadir, "regtest", ".cookie"), 'r', encoding="ascii") as f:
             userpass = f.read()
             split_userpass = userpass.split(':')
             user = split_userpass[0]
             password = split_userpass[1]
+    except FileNotFoundError:
```
Comment (MarcoFalke) [inline test/functional/test_framework/util.py]:
```suggestion
    except OSError:
```

To also catch PermissionError, to not regress on 2753285be72dd051cf2987ca5c80d0b149914020?

See https://docs.python.org/3/library/exceptions.html#os-exceptions

**Pull request merged**